core/process.py

```diff
@@ -102,29 +104,31 @@ def RemoveUserTrap(self, sig_num):
 class _FdFrame(object):
   def __init__(self):
     # type: () -> None
-    self.saved = []  # type: List[Tuple[int, int]]
-    self.need_close = []  # type: List[int]
+    self.saved = []  # type: List[Tuple[int, int, bool]]
     self.need_wait = []  # type: List[Tuple[Process, Waiter]]
 
   def Forget(self):
     # type: () -> None
     """For exec 1>&2."""
+    for saved, orig, forget in reversed(self.saved):
+      if saved != NO_FD and forget:
+        posix.close(saved)
+
     del self.saved[:]  # like list.clear() in Python 3.3
-    del self.need_close[:]
     del self.need_wait[:]
 
   def __repr__(self):
     # type: () -> str
-    return '<_FdFrame %s %s>' % (self.saved, self.need_close)
+    return '<_FdFrame %s>' % self.saved
 
 
 class FdState(object):
   """This is for the current process, as opposed to child processes.
 
   For example, you can do 'myfunc > out.txt' without forking.
   """
-  def __init__(self, errfmt, job_state):
-    # type: (ErrorFormatter, JobState) -> None
+  def __init__(self, errfmt, job_state, mem = None):
```

Comment:
mem=None here

Reply:
Thank you! I'm sorry, this was an oversight. Fixed 704ba1ec.